<launch>
  
  <!-- required inputs when calling file -->
  <arg name="robot_ip" />
  <arg name="robot_dof" />

  <!-- calls the ezgripper_control file -->
  <node name="ezgripper_control" pkg="gripper_control" type="ezgripper_control.py" launch-prefix="gnome-terminal --command" />

  <!-- calls the original xarm_planner_file -->
  <include file="$(find xarm_planner)/launch/xarm_planner_realHW.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="robot_dof" value="6" />
    <arg name="add_gripper" value="true" />
  </include>
  
  <!-- node name="control_server_node" pkg="arm_control" type="control_server_node" output="screen" / -->
  <node name="control_server_node" pkg="arm_control" type="control_server_node" launch-prefix="gnome-terminal --command" />
  <node name="control_client_node" pkg="arm_control" type="control_client_node" launch-prefix="gnome-terminal --command" />

</launch>